common/djangoapps/student/models.py

```diff
@@ -1793,6 +1793,11 @@ def enforce_single_login(sender, request, user, signal, **kwargs):    # pylint:
         else:
             key = None
         if user:
+            if not UserProfile.objects.filter(user=user).exists():
+                # if user has not profile then create it.
+                user_profile = UserProfile(name=user.username, user=user)
+                user_profile.save()
+
```

Comment:
@doctoryes 

> Right - so this added UserProfile creation takes care of the case you describe. The other UserProfile creation is the one which handles creation in the typical case. Correct?

yes

> Regarding the second:
The get_or_create method is a convenience method for exactly the situation you're coding. The docs are here:
https://docs.djangoproject.com/en/1.9/ref/models/querysets/#django.db.models.query.QuerySet.get_or_create
It handles the creation of a non-existent record in a single-line, standard way. It'd be nice if the UserProfile creation to which you referred above was also converted to this method - to standardize both creations.

My concern is UserProfile.name is not constant filed , if user changes his name from profile the in 
``user_profile, __ = UserProfile.objects.get_or_create(name=user.username, user=user)``  => In this scenario it is possible that
- UserProfile.name is not equal  to user.username 
- UserProfile.user is equal to user

because user can change UserProfile.name any time, in that case ``user_profile, __ = UserProfile.objects.get_or_create(name=user.username, user=user)`` may create new object